### PR TITLE
feat: add CI shellcheck

### PR DESCRIPTION
**Pull request by mtulio**
Add basic shellcheck to CI

Ref https://github.com/openshift/release/pull/28245

### CODE REVIEW

File: hack/shellcheck.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
```
Comment (mtulio) [inline hack/shellcheck.sh:1]: This is "as it's" from https://github.com/openshift/cluster-monitoring-operator/blob/master/hack/shellcheck.sh, also used in the our CI

**Comment by mtulio** [comment]
ptal?
/cc @bostrt

**Comment by bostrt** [comment]
Should this PR be with mvp branch instead of main?

**Comment by mtulio** [comment]
> Should this PR be with mvp branch instead of main?

@bostrt it makes more sense as we have no plans for now to merge the entire `mvp` to main.

Changed the base, ptal?

**Review by bostrt — APPROVED** [review]

**Pull request merged**